api/test/baggage/propagation/baggage_propagator_test.cc

```diff
@@ -0,0 +1,68 @@
+#include "opentelemetry/baggage/propagation/baggage_propagator.h"
+#include <gtest/gtest.h>
+#include <map>
+#include <string>
+
+using namespace opentelemetry;
+using namespace opentelemetry::baggage::propagation;
+
+class BaggageCarrierTest : public context::propagation::TextMapCarrier
+{
+public:
+  BaggageCarrierTest() = default;
+  virtual nostd::string_view Get(nostd::string_view key) const noexcept override
+  {
+    auto it = headers_.find(std::string(key));
+    if (it != headers_.end())
+    {
+      return nostd::string_view(it->second);
+    }
+    return "";
+  }
+  virtual void Set(nostd::string_view key, nostd::string_view value) noexcept override
+  {
+    headers_[std::string(key)] = std::string(value);
+  }
+
+  std::map<std::string, std::string> headers_;
+};
+
+static BaggagePropagator format;
+
+TEST(BaggagePropagatorTest, ExtractNoBaggageHeader)
+{
+  BaggageCarrierTest carrier;
+  carrier.headers_      = {};
+  context::Context ctx1 = context::Context{};
+  context::Context ctx2 = format.Extract(carrier, ctx1);
+  auto ctx2_baggage     = baggage::propagation::GetBaggage(ctx2);
+  EXPECT_EQ(ctx2_baggage->ToHeader(), "");
+}
+
+TEST(BaggagePropagatorTest, ExtractAndInjectBaggage)
+{
+  std::string max_baggage_kv = std::string(baggage::Baggage::kMaxKeyValueSize / 2 + 1, 'k') + "=" +
```

Comment:
What is the `+1` for?

Reply:
This is to ensure that header size is larger than allowed for baggage (kMaxKeyValueSize) and hence test the case for large baggage headers. I have now added comment and changed the variable name to make it more clear.